docs/examples/crypto-sockets.md

```diff
@@ -0,0 +1,160 @@
+- Use algorithms to trade crypto and profit. Just kidding, this is left to an exercise for the reader.
+
+To start off, we are going to diverge into some concepts around markets, exchanges and orders.
+
+A Limit Order Book, or just Order Book is a record of all limit orders that are made. a limit order is an order to buy (bid) or sell (ask) an asset for a given price. This could facilitate the exchange of dollars for shares or, as in our case, they could be orders to exchange crypto currencies. On exchanges, the limit order book is constantly changing as orders are placed every fraction of a second. The order book can give a trader insight into the market, whether they are looking to determine liquidity, to create liquidity, design a trading algorithm or maybe determine when bad actors are trying to manipulate the market. In the order book, the **ask price** is the lowest price that a seller is willing to sell at and the **bid price** is the highest price that a buyer is willing to buy at. A limit order is different than a market order in that the limit order can be placed with generally 4 dimensions, the direction (buy/sell), the size, the price and the duration (time to expire). A market order, in comparison, has 2 dimensions, the direction and the size. It is up to the exchange to fill a market order and it is filled via what is available in the order book.
+
+An exchange will generally offer a few different tiers of information that traders can subscribe to. These can be quite expensive for some exchanges, but luckily for us, most crypto exchanges provide access to the various levels for free! For maintaining our order book we are going to need at least level2 order information. This gives us granularity of each limit order so that we can adjust the book in real-time. Without the ability to maintain our own order book, the snapshot would get almost instantly out of date and we would not be able to act with perfect information. We would have to download the order book every millisecond, or faster, to stay up to date and since the order book can be quite large, this isn't really feasible.
+
+Alright, let's get started!
+
+### Inputs & Outputs
+
+We are going to use the `spawn_cluster` approach in this dataflow, this allows us to scale our dataflow in the future if we add additional products that we would like to track. To begin, we will need to build an input builder for our dataflow. In  the input builder we will receive each new message. To start, we use the coinbase pro websocket url (`wss://ws-feed.pro.coinbase.com`) and the Python websocket library to create a connection. Once connected we can send a message to the websocket subscribing to product_ids (pairs of currencies - USD/BTC for this example) and channels (level2 order book data). Finally since we know there will be some sort of acknowledgement message we can grab that with `ws.recv()` and print it out.
+
+```Python
+from websocket import create_connection
```

Comment:
What are the pip packages you need to run this locally? Maybe comment on that in the text? I managed to get it running with `pip install websocket-client`.